hw3/task2/task2/Trie.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace task2
+{
+    /// <summary>
+    /// search tree, key-value data structure with string keys and integer values
+    /// </summary>
+    public class Trie
+    {
+        private class Node
+        {
+            public Dictionary<char, Node> children = new();
+            public bool isKey = false;
+            public int keysNumberInSubtree = 0;
+            public int value = -1;
+        }
+        private Node root = new();
+        /// <summary>
+        /// number of keys in the Trie
+        /// </summary>
+        public int Size { get { return root.keysNumberInSubtree; } }
```

Comment:
```suggestion
        public int Size => root.keysNumberInSubtree;
```
Это в точности то же самое, но короче